runtime/worker.rs

```diff
@@ -769,6 +803,7 @@ impl MainWorker {
   ///
   /// Does not poll event loop, and thus not await any of the "unload" event handlers.
   pub fn dispatch_unload_event(&mut self) -> Result<(), AnyError> {
+    self.dispatch_process_exit_event()?;
```

Comment:
I don't think this is correct, we should probably dispatch it after dispatching Web event